src/state/pages/reducer.js

```diff
@@ -1,19 +1,108 @@
-import { SET_VIEWPAGES } from 'state/pages/types';
+import { combineReducers } from 'redux';
+import {
+  ADD_PAGES,
+  TOGGLE_PAGE_EXPANDED,
+  SET_PAGE_LOADING,
+  SET_PAGE_LOADED,
+  SET_VIEWPAGES,
+} from 'state/pages/types';
 
-const initialState = {
-  viewPages: [],
+// creates a map from an array
+const toMap = (array, propKey) => array.reduce((acc, page) => {
+  acc[page.code] = propKey ? page[propKey] : page;
+  return acc;
+}, {});
+
+// map reducer
```

Comment:
can we use a more meaningful name for the reducer (e.g. `pagesMap`) and remove the comment?